packages/medusa/src/api/store/carts/[id]/route.ts

```diff
@@ -37,11 +52,23 @@ export const POST = async (
     },
   })
 
+  const withInventoryQuantity =
+    req.queryConfig.fields.some((field) => field.includes("items.variant")) &&
+    req.queryConfig.fields.some((field) => field.includes("items.variant.inventory_quantity"))
```

Comment:
nit: we don't need both checks here
```suggestion
  const withInventoryQuantity =
    req.queryConfig.fields.some((field) => field.includes("items.variant.inventory_quantity"))
```

Reply:
I removed the double check, but this causes an issue. When using `fields=+items.variant.inventory_quantity`, the helper filters out `items.variant.inventory_quantity` but doesn't ensure `items.variant` is fetched, resulting in `cart.items[x].variant` being undefined and causing this error:

```
TypeError: Cannot read properties of undefined (reading 'id')
at wrapVariantsWithInventoryQuantityForSalesChannel
```

Should we ensure that `fields=+items.variant.inventory_quantity` automatically adds `items.variant` ?